### ISSUE TITLE
Incorrect accessibility for local definitions, parameters and type parameters

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#1325]
Currently, if a definition is not private or protected, it's assigned with `Some(s.Accessibility(PUBLIC))`. This is incorrect, because things like local vals or pattern variables in expressions cannot be public. I would say that non-global symbols must have their `accessibility` set to `None`.

There's also a question of whether it is appropriate to say that some global symbols (namely, `PARAMETER` and `TYPE_PARAMETER` symbols) have accessibility. The spec is silent on that matter too - it neither lists valid accessibilities for parameters and type parameters, nor says that no accessibilities are supported.
